crates/sui-core/src/authority/authority_store.rs

```diff
@@ -137,6 +137,7 @@ impl AuthorityStore {
         indirect_objects_threshold: usize,
         enable_epoch_sui_conservation_check: bool,
         registry: &Registry,
+        load_genesis: bool,
```

Comment:
Why not making `genesis` optional? Is there anything else that download_formal_snapshot needs from genesis?

Reply:
Genesis is needed elsewhere, for example to construct the `EpochStartConfiguration`. We can't rely on the presence of data in perpetual store either, because in the case of snapshot restore, perpetual store is empty but we still do not want to load genesis objects

Reply:
When we restore from a snapshot, would it include a correct EpochStartConfiguration?
In anyway I think the point is that we shouldn't include anything from genesis in this case because there is always risk of introducing something too old stored in the snapshot restore process.
Relying on the snapshot + some setup after restoring seems the correct way to go imho

Reply:
This can go in to unblock the fix as it's urgent, but I would prefer we revisit the code path so that the snapshot restoration process does not rely on genesis in anyway possible.

Reply:
Oh I see the point you're making. I can do that here

Reply:
Fixed, PTAL